helm/vertical-pod-autoscaler-app/values.yaml

```diff
@@ -1,189 +1,568 @@
-    pullPolicy: IfNotPresent
-    # recommender.image.tag -- Overrides the image tag whose default is the chart appVersion
-    tag: ""
-  # recommender.podAnnotations -- Annotations to add to the recommender pod
-  podAnnotations: {}
-  # recommender.podLabels -- Labels to add to the recommender pod
-  podLabels: {}
-  # recommender.podSecurityContetxt -- The security context for the recommender pod
-  podSecurityContext:
-    runAsNonRoot: true
-    runAsUser: 65534
-  # recommender.securityContext -- The security context for the containers inside the recommender pod
-  securityContext: {}
-  # recommender.resources -- The resources block for the recommender pod
-  resources:
-    limits:
-      cpu: 200m
-      memory: 3000Mi
-    requests:
-      cpu: 50m
-      memory: 500Mi
-  nodeSelector: {}
-  tolerations: []
-  affinity: {}
-
-updater:
-  # updater.enabled -- If true, the updater component will be deployed
-  enabled: true
-  # updater.extraArgs -- A key-value map of flags to pass to the updater
-  extraArgs:
-    # update.extraArgs.min-replicas -- Minimum number of replicas to perform update
-    min-replicas: 1
-    # update.extraArgs.v -- Verbosity level for updater
-    v: 2
-  replicaCount: 1
-  # updater.podDisruptionBudget -- This is the setting for the pod disruption budget
-  podDisruptionBudget: {}
-    # maxUnavailable: 1
-  image:
-    # updater.image.repository -- The location of the updater image
-    repository: giantswarm/vpa-updater
-    # updater.image.pullPolicy -- The pull policy for the updater image. Recommend not changing this
-    pullPolicy: IfNotPresent
-    # updater.image.tag -- Overrides the image tag whose default is the chart appVersion
-    tag: ""
-  # updater.podAnnotations -- Annotations to add to the updater pod
-  podAnnotations: {}
-  # updater.podLabels -- Labels to add to the updater pod
-  podLabels: {}
-  # updater.podSecurityContetxt -- The security context for the updater pod
-  podSecurityContext:
-    runAsNonRoot: true
-    runAsUser: 65534
-  # updater.securityContext -- The security context for the containers inside the updater pod
-  securityContext: {}
-  # updater.resources -- The resources block for the updater pod
-  resources:
-    limits:
-      cpu: 200m
-      memory: 3000Mi
-    requests:
-      cpu: 50m
-      memory: 500Mi
-  nodeSelector: {}
-  tolerations: []
-  affinity: {}
-
-admissionController:
-  # admissionController.enabled -- If true, will install the admission-controller component of vpa
-  enabled: true
-  # admissionController.extraArgs -- A key-value map of flags to pass to the admissionController
-  extraArgs:
-    webhook-timeout-seconds: 5
-  # admissionController.generateCertificate -- If true and admissionController is enabled, a pre-install hook will run to create the certificate for the webhook
-  generateCertificate: true
-  certGen:
+registry: docker.io
```

Comment:
wow, this new `values.yaml` is huge!

Do we need/use all of these values in our templates? Or did you just take the whole sub-chart's values?
If the latter, think that we'll have to maintain it up-to-date when it changes upstream.
That's why I usually just set the values we override, those where we want to change the defaults. And have a link to the sub-chart syntax in our documentation.

Reply:
You're right, I'll have to cut some parts of the values since most of it is not overridden.

Reply:
Values updated

Reply:
I still think there's too much stuff in there, and it will require extra work during upgrades.
And did you re-generated the schema after updating the values? :grin: